### ISSUE TITLE
Switch

### ISSUE THREAD

**Issue by somethingSTRANGE** [matkoch/resharper-cognitivecomplexity#18]
I'm using Rider 2020.2.4 with CognitiveComplexity 2020.2.2.

The `switch` doesn't appear to be calculated correctly.

According to the linked SonarSource [Cognitive Complexity](https://www.sonarsource.com/docs/CognitiveComplexity.pdf) document, the section on Switches states:

> ### Switches
>A switch and all its cases combined incurs a single structural increment.
>
>Under Cyclomatic Complexity, a switch is treated as an analog to an if-else if chain. That is, each case in the switch causes an increment because it causes a branch in the mathematical model of the control flow.
>
>But from a maintainer’s point of view, a switch - which compares a single variable to an explicitly named set of literal values - is much easier to understand than an if-else if chain because the latter may make any number of comparisons, using any number of variables and values.
>
>In short, an if-else if chain must be read carefully, while a switch can often be taken in at a glance.

Yet, the following method produces a complexity rating of 100% with the `switch` and each `case` getting a "+1". My understanding is that the entire `switch` should add just 10% complexity (a single +1 increment), regardless of the number of `case`  labels under it.
```
private void SwitchMethod(int value)
{
   switch (value)
   {
      case 1: break;
      case 2: break;
      case 3: break;
      case 4: break;
      case 5: break;
      case 6: break;
      case 7: break;
      case 8: break;
      case 9: break;

      default: throw new ArgumentOutOfRangeException();
   }
}
```

Converting the `switch` to a `if-else` currently results in the same 100% complexity rating:
```
private void SwitchMethod(int value)
{
   if (value == 1) { }
   else if (value == 2) { }
   else if (value == 3) { }
   else if (value == 4) { }
   else if (value == 5) { }
   else if (value == 6) { }
   else if (value == 7) { }
   else if (value == 8) { }
   else if (value == 9) { }
   else
   {
      throw new ArgumentOutOfRangeException();
   }
}
```

Is this deviation from Campbell's paper intentional or an oversight?

It always struck me as odd that the `switch` received the same rating as the `if-else`, but I never bothered to find out why. I happened upon Campbell's paper today when reading about problems with _Cyclomatic Complexity_, and only then discovered the section on "Switches", and that this plugin was based on that document.

I suppose when it comes to _pattern matching_ `switch` statements, they might be more cognitively complex, but the standard `switch` statement do seem less complex than their `if-else` counterparts.

**Comment by matkoch** [comment]
To me this seems correct. Switch increases nesting level, switch and break increment complexity value.![image](https://user-images.githubusercontent.com/5005566/100171781-38b9c380-2ec7-11eb-8877-a84154dfc845.png)

**Comment by somethingSTRANGE** [comment]
I actually haven't read the entire document yet, and just saw the appendix now, however that doesn't include the `case` labels associated with the `switch`. The `switch` certainly increments, however the `case` label should not. That's the issue.

There are numerous examples of the `switch` scoring in that document, and it seems like the entire `switch` (`case` labels included) should be treated as a single structural increment, so +1 when not nested, or +2 when nested once, etc.

![chrome_NXiDSh7lr9](https://user-images.githubusercontent.com/6905832/100174767-eefaba00-2e81-11eb-8035-46917dee0cc0.png)

The `switch` has a complexity of +1 in the above example captured from Campbell's paper.

**Comment by matkoch** [comment]
There's a difference between `break` and `return`.

**Comment by somethingSTRANGE** [comment]
That makes sense. I `break` add the complexity not the `case` label.

**Comment by somethingSTRANGE** [comment]
Yep. That's it. I have a method with just a `switch` that has many `case` labels with `break` and the method gets a high complexity score. If I were to change those to `return` the score would drop.

From the document:
> ### Jumps to labels
> `goto`, and `break` or `continue` to a label add fundamental increments to Cognitive Complexity. But because an early `return` can often make code much clearer, no other jumps or early exits cause an increment.

I was mistaken. This is not an issue. Thanks for clearing it up.

**Comment by matkoch** [comment]
FYI, I let this review from the author and she approved all tests back then. If you have any questions/concerns I'd ask you to ping her on Twitter, and only report back here when she agrees. Her account is https://twitter.com/GAnnCampbell.

**Comment by somethingSTRANGE** [comment]
I did already report a issue regarding `?:`, however I'll ping her if I see anything else. Thanks!

**Comment by matkoch** [comment]
Btw, if you click the code vision item, you'll see the places that account for an increment: 
<img width="940" alt="image" src="https://user-images.githubusercontent.com/5005566/100211357-552c1f00-2f0c-11eb-92c0-276c0b21bb8d.png">
